disnake/state.py

```diff
@@ -1679,22 +1689,27 @@ def parse_voice_server_update(self, data) -> None:
                 logging_coroutine(coro, info="Voice Protocol voice server update handler")
             )
 
-    def parse_typing_start(self, data) -> None:
+    def parse_typing_start(self, data: TypingEvent) -> None:
         channel, guild = self._get_guild_channel(data)
+        raw = RawTypingEvent(data)
+
+        user_id = int(data["user_id"])
+        member_data = data.get("member")
+        if member_data and guild is not None:
+            # try member cache first
+            raw.member = guild.get_member(user_id) or Member(
+                data=member_data, guild=guild, state=self
+            )
+
+        self.dispatch("raw_typing", raw)
+
         if channel is not None:
             member = None
-            user_id = utils._get_as_snowflake(data, "user_id")
-            if isinstance(channel, DMChannel):
-                member = channel.recipient
+            if raw.member is not None:
+                member = raw.member
 
-            elif isinstance(channel, (Thread, TextChannel)) and guild is not None:
```

Comment:
Is this case covered in L1698? (just clarifying)

Reply:
Yes, at least if the existence of `data["guild_id"]` also implies the existence of `data["member"]`, which should be a valid assumption if I'm reading the [api docs](https://discord.com/developers/docs/topics/gateway#typing-start) correctly.
(https://discord.com/channels/808030843078836254/913779868985090089/915271250031570944)

Reply:
Alright then